rust/perspective/arrow_accessor/src/lib.rs

```diff
@@ -0,0 +1,102 @@
+/******************************************************************************
+ *
+ * Copyright (c) 2019, the Perspective Authors.
+ *
+ * This file is part of the Perspective library, distributed under the terms of
+ * the Apache License 2.0.  The full license can be found in the LICENSE file.
+ *
+ */
+#![cfg(target_arch = "wasm32")]
+
+pub mod accessor;
+
+use std::io::Cursor;
+
+use arrow::ipc::reader::StreamReader;
+use arrow::record_batch::RecordBatch;
+
+use js_sys::Array;
+use wasm_bindgen::prelude::*;
+
+use crate::accessor::ArrowAccessor;
+
+#[wasm_bindgen]
+extern "C" {
+    #[wasm_bindgen(js_namespace = console)]
+    pub fn log(s: &str);
+}
+
+// Debug `panic!` messages inside console.error().
+pub fn set_panic_hook() {
+    #[cfg(feature = "console_error_panic_hook")]
+    console_error_panic_hook::set_once();
+}
+
+/// Load an arrow binary in stream format.
+pub fn load_arrow_stream(buffer: Box<[u8]>) -> Box<ArrowAccessor> {
+    let cursor = Cursor::new(buffer);
+    let reader = StreamReader::try_new(cursor).unwrap();
+    let schema = reader.schema();
+
+    // Iterate over record batches, and collect them into a vector of
+    // heap-allocated batches.
+    let batches = reader
+        .map(|batch| {
+            // Panic if it can't read the batch
+            Box::new(batch.unwrap())
+        })
+        .collect::<Vec<Box<RecordBatch>>>();
+
+    if let [batch] = &batches[..] {
+        Box::new(ArrowAccessor::new(batch.clone(), schema))
+    } else {
+        panic!("Arrow should only contain a single record batch.")
```

Comment:
This is because Perspective exports Arrows as a single record batch, so I figured there was no need for joining multiple batches into a single array etc.